neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/transaction/impl/Neo4jTransactionImplementor.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.transaction.impl;
+
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+
+import org.hibernate.engine.transaction.internal.jta.JtaStatusHelper;
+import org.hibernate.engine.transaction.jta.platform.spi.JtaPlatform;
+import org.hibernate.engine.transaction.spi.TransactionImplementor;
+import org.hibernate.ogm.datastore.neo4j.impl.Neo4jDatastoreProvider;
+import org.hibernate.ogm.datastore.neo4j.logging.impl.Log;
+import org.hibernate.ogm.datastore.neo4j.logging.impl.LoggerFactory;
+import org.hibernate.ogm.transaction.impl.ForwardingTransactionImplementor;
+import org.neo4j.graphdb.Transaction;
+
+/**
+ * @author Davide D'Alto
+ */
+public class Neo4jTransactionImplementor extends ForwardingTransactionImplementor {
```

Comment:
Could you add a comment about the motivation for having this class? The reader should know what's special about it.